src/routes/notifications/v1/notifications.controller.ts

```diff
@@ -47,24 +48,37 @@ export class NotificationsController {
   async registerDevice(
     @Body() registerDeviceDto: RegisterDeviceDto,
   ): Promise<void> {
-    await this.notificationsService.registerDevice(registerDeviceDto);
-
     if (this.isPushNotificationV2Enabled) {
       // Compatibility with V2
-      // @TODO Remove NotificationModuleV2 after all clients have migrated and compatibility is no longer needed.
       const compatibleV2Requests =
         await this.createV2RegisterDto(registerDeviceDto);
 
       const v2Requests = [];
+
       for (const compatibleV2Request of compatibleV2Requests) {
         v2Requests.push(
           await this.notificationServiceV2.upsertSubscriptions(
             compatibleV2Request,
           ),
         );
       }
-
       await Promise.all(v2Requests);
+
+      // Remove tokens from the old service to prevent duplication.
+      if (registerDeviceDto.uuid) {
+        const unregistrationRequests = [];
+        for (const safeRegistration of registerDeviceDto.safeRegistrations) {
+          unregistrationRequests.push(
+            this.notificationsService.unregisterDevice({
+              chainId: safeRegistration.chainId,
+              uuid: registerDeviceDto.uuid,
+            }),
+          );
+        }
+        await Promise.all(unregistrationRequests).catch(() => {}); // If the device is not already registered, the TX service will throw a 404 error, but we can safely ignore it.
```

Comment:
Regarding logging the errors, my concern is that once the token is removed from the TX service, any subsequent calls will throw errors, potentially resulting in a flood of error logs in a short while.
I don’t have a strong opinion on using allSettled, as in the worst case, it would result in delivering notifications twice to more devices. I mean, it only reduces the likelihood since less Promises are rejected.

Reply:
Changed `Promise.all` to `Promise.allSettled`
https://github.com/safe-global/safe-client-gateway/pull/2173/commits/09f5a7ac28900345eafc849381a919253ebda773